hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/block/SCMBlockDeletingService.java

```diff
@@ -248,4 +248,9 @@ public boolean shouldRun() {
   public String getServiceName() {
     return SCMBlockDeletingService.class.getSimpleName();
   }
+
+  @Override
+  public void stop() {
```

Comment:
A shutdown thread pool can not be reused again, since after calling shutdown on a ExecutorService, no new Task will be accepted.

Reply:
Yes., but only if the service is stopped.

I think , we don't need to stop the background services while reloading the SCM state during install snapshot, as reload will only happen on follower and on followers, these won't do anything on the db. What do you think?


However, a start and stop method in such services are desired so that, all the registered services can be started and stopped via service manager itself.

Reply:
> What do you think?
I agree.

> However, a start and stop method in such services are desired so that, all the registered services can be started and stopped via service manager itself.
In order to support safely stop() then start(), these service should be rewritten to avoid use ExecutorService. This can be done as a newbie task in future.